include/dlaf/permutations/general/impl.h

```diff
@@ -126,9 +129,51 @@ void applyPermutations(GlobalElementIndex out_begin, GlobalElementSize sz, SizeT
   }
 }
 
+template <class T, Device D, Coord C>
+void applyPermutationsFiltered(
+    GlobalElementIndex out_begin, GlobalElementSize sz, SizeType in_offset,
+    const matrix::Distribution& subm_dist, const SizeType* perm_arr,
+    const std::vector<pika::shared_future<matrix::Tile<const T, D>>>& in_tiles_fut,
+    const std::vector<matrix::Tile<T, D>>& out_tiles, std::function<bool(SizeType)> filter) {
```

Comment:
I realize the direction is arbitrary, so I don't feel too strongly about this, except that it should written in the documentation explicitly whether the predicate filters in or out.

My suggestion would be to rename `filter` to `pred` and have it filter items out when the predicate returns `false`. This would be consistent with https://en.cppreference.com/w/cpp/ranges/filter_view. However, I can see why the current option might be more intuitive as well, which is why the first part of this comment is the important one to avoid confusion.

Reply:
It's a super good point.

I thought it was a good idea to check what other public APIs do...I looked at other functional approaches (e.g. [haskell](https://hackage.haskell.org/package/base-4.18.0.0/docs/Prelude.html#v:filter), [python](https://docs.python.org/3/library/functions.html?highlight=filter#filter)) and then I forgot to do apply the change 😛 

I fully agree with you to filter out when false. I'll do it.